### DISCUSSION TITLE
What is the expected impact of Hashicorp adopting the BSL?

### DISCUSSION THREAD

**Discussion by stevehipwell** [crossplane/crossplane#4483] [General]
With the announcement that Hashicorp are [adopting the BSL](https://www.hashicorp.com/blog/hashicorp-adopts-business-source-license) I'm interested if this is going to be an issue for the Crossplane providers developed with Upjet from official Hashicorp Terraform providers (e.g. AWS, GCP & Azure)? Could it also be an issue for the Crossplane Terraform provider? The following statement (taken from the link above) makes it sound like Hashicorp intend to block this kind of behaviour.

>Vendors who provide competitive services built on our community products will no longer be able to incorporate future releases, bug fixes, or security patches contributed to our products.

**Comment by jbw976** [comment]
Good question @stevehipwell! @bassam took the time today to write down his thoughts and more information in this post: https://bit.ly/47qP3OX

In short, this licensing change does not impact the project's use of Terraform.
